### ISSUE TITLE
OmniSharp.exe using Windows Subsystem for Linux

### ISSUE THREAD

**Issue by frondeus** [OmniSharp/omnisharp-emacs#316]
I wonder, if I could use emacs25 on WSL (Bash on Ubuntu on Windows) (it's working OOB after Creators Update) with OmniSharp-Roslyn.

I know that
> net46 build requires mono (>=4.8.0) if it is run on a platform other than Windows.
And I want to avoid this by using native OmniSharp-Roslyn.

After Creators Update we gained
> Windows <-> Linux Interop

So my question is - can I use it? 

I tried to use scripts like:
```
#!/bin/sh
set -e
/mnt/c/omnisharp/OmniSharp.exe "$@"
```
and
```
#!/bin/sh
set -e # based on Cyngwin script.
/mnt/c/omnisharp/OmniSharpWrapper.exe "C:\omnisharp\OmniSharp.exe" "$@"
```

I also tried 
```
(setq omnisharp-server-executable-path "/mnt/c/omnisharp/OmniSharp.exe")
```

I can start OmniSharp manually and it's listening on 2000, so maybe I could just attach emacs to it (but I'm not sure how, I have errors like `Start it with 'omnisharp-start-omnisharp-server' first`).

I know I can use cyngwin (and it's working) but ... WSL seems to be fun!

**Comment by razzmatazz** [comment]
@frondeus not sure on which host are you running emacs? native windows or the one in linux subsystem?

Also, if you are running in linux subsystem, could you try the M-x `omnisharp-install-server` command which has recently been implemented that would attempt to install the server automatically (which should work as it would think emacs is running on Windows).

And then yes, you need to have mono installed on your linux subsystem for this `omnisharp-install-server` command to work as the server being downloaded is mono version.

**Comment by razzmatazz** [comment]
And you don't need http, I think. Unless you are attempting to run emacs on linux subsystem and the server natively on Windows, or vice versa..

**Comment by frondeus** [comment]
I attempt to run Emacs on linux subsystem and Omnisharp server natively on Windows.

I know I could use Omnisharp server on linux but my project must be on windows filesystem (for example `C:/Code`) and for this moment accessing it from linux subsystem is rather inefficient [uservoice](https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/suggestions/13386252-filesystem-slow-compared-with-native-ubuntu-system).

Now, Omnisharp server on Windows would be better solution - it has no filesystem performance issues.

**Comment by razzmatazz** [comment]
yeah, you might need path translations working in emacs for this to work... not sure if it would work for you

there has been some work done on cygwin where paths are translated from unix-to-windows and back but this "bash on windows" is probably another story which I don't think would work.. you can try, though..

so.. why can't you run both emacs and server on windows? just extract https://github.com/OmniSharp/omnisharp-roslyn/releases/download/v1.21.0/omnisharp-win-x64-net46.zip somewhere and then point `omnisharp-server-executable-path` to where it needs to be on windows, natively?

**Comment by razzmatazz** [comment]
oops, sorry, don't use 64bit binary for now! made my own mistake.. it crashes emacs for now, see https://github.com/OmniSharp/omnisharp-emacs/issues/315

use the 32bit version on windows host, https://github.com/OmniSharp/omnisharp-roslyn/releases/download/v1.21.0/omnisharp-win-x86-net46.zip

**Comment by razzmatazz** [comment]
@frondeus also, we now have `M-x omnisharp-install-server` working on Windows (this requires powershell v5+). could you check if that works for you? so you wouldn't neet to mess with manual setting up of binary paths

also, if you use `omnisharp-install-server` then you should remove `omnisharp-server-executable-path` from your init.el
